cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -550,14 +533,28 @@ public void saveProgramRuntimeArgs(HttpRequest request, HttpResponder responder,
                                      @PathParam("program-type") String type,
                                      @PathParam("program-name") String programName) throws Exception {
     ProgramType programType = getProgramType(type);
+    ProgramId programId = new ApplicationId(namespaceId, appName, appVersion).program(programType, programName);
+    saveProgramIdRuntimeArgs(programId, request, responder);
+  }
+
+  private void getProgramIdRuntimeArgs(ProgramId programId, HttpResponder responder) throws BadRequestException,
+    NotImplementedException, NotFoundException, UnauthorizedException {
+    ProgramType programType = programId.getType();
     if (programType == null || programType == ProgramType.WEBAPP) {
```

Comment:
The error also carries "saving/getting" based on the call. Don't think it helps to create a function call to save one common line of code.